### PR TITLE
Adding dtype test to main

### PR DESCRIPTION
**Pull request by ronniepiku**
# Description

A dummy env was created. And then test we're added to identify the supported dtypes.

Fixes # (issue), Depends on # (pull request)

## Type of change

> Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

### Screenshots

# Checklist:

- [x ] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [ ] I have run `pytest -v` and no errors are present.
- [x ] I have commented my code, particularly in hard-to-understand areas
- [ x] I have made corresponding changes to the documentation
- [ ] I solved any possible warnings that `pytest -v` has generated that are related to my code to the best of my knowledge.
- [x ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/test_supported_dtypes.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import gym
```
Comment (rodrigodelazcano) [inline tests/test_supported_dtypes.py:1]:
```python
import gymnasium as gym
```

**Review by rodrigodelazcano — CHANGES_REQUESTED** [review]

File: tests/test_supported_dtypes.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    if unsupported_dtypes:
+        print(
+            "Unsupported dtypes for Discrete space:",
+            [dtype.__name__ for dtype in unsupported_dtypes],
+        )
+    else:
+        print("All dtypes are supported for Discrete space")
+
+
+test_box_dtypes()
```
Comment (balisujohn) [inline tests/test_supported_dtypes.py:97]: these calls are unnecessary for tests

File: minari/serialization.py [hunk 3]
```diff
@@ -18,11 +18,11 @@ def serialize_space(space: spaces.Space, to_string=True) -> Union[Dict, str]:
 def _serialize_box(space: spaces.Box, to_string=True) -> Union[Dict, str]:
     result = {}
```
Comment (balisujohn) [inline minari/serialization.py:19]: what is the rationale behind these changes to serialization.py?

**Review by balisujohn — CHANGES_REQUESTED** [review]
Thanks for your work on this so far. I requested some changes and asked a clarifying question. I also think it would be good to add a table to the `dataset_standards.md` page it the doc clarifying which dtypes are supported for each space. Also, I noticed tests are failing, definitely please check that 

````
python3 -m pytest
````
and
````
pre-commit run --all-files
````
run without any issues locally.

File: tests/common.py [hunk 4]
```diff
@@ -477,10 +541,77 @@ def check_data_integrity(data: MinariStorage, episode_indices: Iterable[int]):
                 episode["observations"], i
             )
             assert data.observation_space.contains(obs)
+            validate_component_dtype(
```
[the remark this answers is not in the document]
Comment (ronniepiku) [reply tests/common.py:544]: Looking back now, it seems that check_data_integrity is usesless so I have deleted it and used validate_component_dtype instead for the test. I also think this should take arguments of action_space, observation_space and expected_dtype. @balisujohn

**Pull request closed without merging**